jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js

```diff
@@ -108,6 +110,134 @@ YUI.add('user-profile-model-list', function() {
+        action: this._displayConfirmation.bind(this, null),
+        type: 'base'
+      }, {
+        title: 'Destroy',
+        action: this._destroyModel,
+        type: 'destructive'
+      }];
+      const name = model.name.split('/')[1];
+      const message = `Are you sure you want to destroy ${name}?`
+        + ' All the applications and units included in the model will be'
+        + ' destroyed. This action cannot be undone.';
+      return (
+        <juju.components.ConfirmationPopup
+          buttons={buttons}
+          message={message}
+          title="Destroy model" />);
+    },
+
+    /**
+      Makes a request of JIMM or JES to delete a selected model.
+
+      @method _destroyModel
+    */
+    _destroyModel: function() {
+      const model = this.state.destroyModel;
+      const uuid = model.uuid;
+      // Hide the confirmation popup.
+      this._displayConfirmation(null);
+      const xhr = this.props.destroyModels([uuid], this._destroyModelCallback);
+      this.xhrs.push(xhr);
+    },
+
+    /**
+      Callback for the JIMM and JES delete model call.
+
+      @method _destroyModelCallback
+      @param {String} err The error from the request, or null.
+      @param {Array} modelList The list of models.
+    */
+    _destroyModelCallback: function(err, results) {
+      const addNotification = this.props.addNotification;
+      // Handle global errors.
+      if (err) {
+        addNotification({
+          title: 'Model destruction failed',
+          message: 'The model failed to be destroyed: ' + err,
+          level: 'error'
+        });
+        return;
+      }
+      // Check for bad data.
+      let uuids;
+      try {
+        uuids = Object.keys(results);
+      } catch (e) {
+        addNotification({
+          title: 'Model destruction failed',
+          message: 'Bad data returned from model destruction.',
+          level: 'error'
+        });
+        return;
+      }
+      if (!uuids || !uuids.length || uuids.length !== 1) {
+        addNotification({
+          title: 'Model destruction failed',
+          message: 'Bad data returned from model destruction.',
+          level: 'error'
+        });
+        return;
+      }
+      const uuid = uuids[0];
+      // Check for model-specific errors.
+      const uuidError = results[uuid];
+      if (uuidError) {
+        addNotification({
+          title: 'Model destruction failed',
+          message: 'The model failed to be destroyed: ' + uuidError,
+          level: 'error'
+        });
+        return;
+      }
+      // Handle a successful deletion.
+      addNotification({
+        title: 'Model destroyed',
+        message: 'The model is currently being destroyed.',
+        level: 'important'
+      });
+      // XXX kadams54: Ideally the model would change in the DB and that would
+      // trigger a re-render. Since we're getting the models from an API
+      // rather than a DB, I'm not sure what the best way is to do this.
```

Comment:
Talked it over with Jeff and we agreed to keep the hack and comment in there for the time being. Once models get re-jiggered to move off YUI, we can re-visit this code.